nzedb/Binaries.php

```diff
@@ -1330,6 +1338,20 @@ protected function _retrieveBlackList($groupName)
 	}
 
 	/**
+	 * Update the last_activity date on the blacklist
+	 * @param int $blackListId The ID of the blacklist entry
+	 * @return void
+	 */
+	public function updateBlacklistActivity()
+	{
+		foreach ($this->_binaryblacklistIdsToUpdate as $bid) {
+			$this->_pdo->queryExec(sprintf('UPDATE binaryblacklist set last_activity = NOW() where id = %d', $bid));
+		}
```

Comment:
Instead of looping, maybe concatenate all the id's in a string, it would make less queries.

` WHERE id IN (%s)', implode(',', $this->_binaryblacklistIdsToUpdate)`